src/compiler/binder.rs

```diff
@@ -146,10 +145,10 @@ impl BinderType {
         self.Symbol()(flags, name).into()
     }
 
-    fn add_declaration_to_symbol<TNode: NodeInterface>(
```

Comment:
I got rid of this `<TNode: NodeInterface>` pattern everywhere since it tends to "infect" other code and I think it's cleaner to just always pass around `&Node` and then call `.as_*()` to get the "unwrapped" variants as needed (and even for functions that don't "infect" others, it's clunkier to pass an `Rc<Node>` (eg `&*node` just `&node`))